Datadog.Serverless.Compat/DatadogServerlessCompat.cs

```diff
@@ -0,0 +1,184 @@
+        }
+
+        private static string GetBinaryPath()
+        {
+            if (!string.IsNullOrEmpty(BinaryPath))
+            {
+                _logger.LogDebug("Detected user configured binary path {BinaryPath}", BinaryPath);
+                return BinaryPath;
+            }
+
+            if (IsWindows())
+            {
+                _logger.LogDebug("Detected {OS}", OS);
+                return "datadog/bin/windows-amd64/datadog-serverless-compat.exe";
+            }
+            else
+            {
+                _logger.LogDebug("Detected {OS}", OS);
+                return "datadog/bin/linux-amd64/datadog-serverless-compat";
+            }
+        }
+
+        private static string SetPackageVersion()
+        {
+            try
+            {
+                var assembly = Assembly.GetExecutingAssembly();
+                var version = assembly
+                    .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                    ?.InformationalVersion;
+                return version ?? "unknown";
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Unable to identify package version");
+                return "unknown";
+            }
+        }
+
+        public static void Start()
+        {
+            var environment = GetEnvironment();
+            _logger.LogInformation("Environment detected: {Environment}", environment);
+
+            if (environment == CloudEnvironment.Unknown)
+            {
+                _logger.LogError(
+                    "{Environment} environment detected, will not start the Datadog Serverless Compatibility Layer",
+                    environment
+                );
+                return;
+            }
+
+            if (!IsWindows() && !IsLinux())
+            {
+                _logger.LogError(
+                    "Platform {OS} detected, the Datadog Serverless Compatibility Layer is only supported on Windows and Linux",
+                    OS
+                );
+                return;
+            }
+
+            var binaryPath = GetBinaryPath();
+            _logger.LogDebug("Spawning process from binary at path {binaryPath}", binaryPath);
+
+            if (!File.Exists(binaryPath))
+            {
+                _logger.LogError(
+                    "Serverless Compatibility Layer did not start, could not find binary at path {binaryPath}",
+                    binaryPath
+                );
+                return;
+            }
+
+            var packageVersion = SetPackageVersion();
+            _logger.LogDebug("Found package version {packageVersion}", packageVersion);
+
+            try
+            {
+                var startInfo = new ProcessStartInfo
+                {
+                    FileName = binaryPath,
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
```

Comment:
No need to redirect these if we're not going to access them.
```suggestion
```

Reply:
Updated here: https://github.com/DataDog/datadog-serverless-compat-dotnet/commit/118ddd4bcafc3ac54b1acae36a59b22f28cdb46f